packages/compass-signtool/src/ssh-client.ts

```diff
@@ -0,0 +1,108 @@
+import { Client, SFTPWrapper } from 'ssh2';
+import { readFileSync } from 'fs';
+import { debug } from './utils';
+
+type SSHClientOptions = {
+  host: string;
+  port: number;
+  username: string;
+  privateKey: string;
+};
+
+export class SSHClient {
+  private sshConnection: Client;
+  private sftpConnection?: SFTPWrapper;
+
+  private connected = false;
+
+  constructor(private sshClientOptions: SSHClientOptions) {
+    this.sshConnection = new Client();
+    this.setupEventListeners();
+  }
+
+  setupEventListeners() {
+    this.sshConnection.on('ready', () => {
+      debug('SSH: Connection established');
+      this.connected = true;
+    });
+    this.sshConnection.on('error', (err) => {
+      debug('SSH: Connection error', err);
+      this.connected = false;
+    });
+    this.sshConnection.on('close', () => {
+      debug('SSH: Connection closed');
+      this.connected = false;
+      this.sshConnection.destroy();
+    });
+  }
+
+  async connect() {
+    if (this.connected) {
+      return Promise.resolve();
+    }
+    return new Promise(async (resolve, reject) => {
+      this.sshConnection.connect({
+        ...this.sshClientOptions,
+        privateKey: readFileSync(this.sshClientOptions.privateKey),
```

Comment:
this is a really silly and optional one: can we use async I/O everywhere? Nowadays is so easy that we can just do it by default and avoid having code that mixes both sync and async, in case we would end up replicating or copy/pasting fragments of code and accidentally introduce sync calls where we don't want to.